pkg/processor/aws_event_bridge.go

```diff
@@ -0,0 +1,189 @@
+package processor
+
+import (
+	"encoding/json"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/eventbridge"
+	"github.com/pkg/errors"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/connection"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/events"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	authMethodAWS  = "access_key"    // only this method is supported by the processor
+	resyncInterval = time.Minute * 5 // resync rule patterns after interval
+)
+
+// awsEventBridge implements the Processor interface
+type awsEventBridge struct {
+	session *session.Session
+	*eventbridge.EventBridge
```

Comment:
also the pointer can be removed I think, on both this and `session`

Reply:
> can we name this `bridge` ? So `*eventbridge.EventBridge` -> `bridge *eventbridge.EventBridge`

So you don't want the embedding here? Gets pretty verbose then, e.g. `eventBridge.bridge.PutEvents`

Reply:
> * If you'd like to leave the struct not accessible outside the package you can go with `vCenterStruct` or `vCenterType` or `vCenterProcessor` or  in general `structOriginalName` + `Interface`.
I like that proposal `xProcessor` (don't export for now, because we return interfaces anyways)

> * Avoid single char named variables, as it can annoy contributors who might want to extend your code
Agree

> * Pointers can be confusing, so [avoid when you can](https://github.com/golang/go/wiki/CodeReviewComments#pass-values). It might be worth it if structs are big enough though.
Changed most of them to value. Are you also referring to pointer semantics on struct methods?

Reply:
I agree, it would indeed get pretty verbose so it is good as it is 👍